modules/channel-web/src/backend/api.ts

```diff
@@ -119,23 +124,25 @@ export default async (bp: typeof sdk, db: Database) => {
     const { botId } = req.params
     const { conversationId, webSessionId } = req.body || {}
 
-    const userId = await bp.realtime.getVisitorIdFromGuestSocketId(webSessionId)
-    if (!userId) {
+    req.visitorId = await bp.realtime.getVisitorIdFromGuestSocketId(webSessionId)
+    if (!req.visitorId) {
       return next(ERR_BAD_USER_SESSION_ID)
     }
 
-    if (!userIdIsValid(userId)) {
+    if (!userIdIsValid(req.visitorId)) {
       return next(ERR_USER_ID_INVALID)
     }
 
-    if (isInt(conversationId)) {
-      const parsedConversationId = parseInt(conversationId)
+    req.messaging = await db.getMessagingClient(botId)
+    const userId = await db.mapVisitor(botId, req.visitorId, req.messaging)
 
-      if (!(await db.isValidConversationOwner(userId, parsedConversationId, botId))) {
+    if (conversationId) {
+      const conversation = await req.messaging.conversations.get(conversationId)
```

Comment:
This is another place where an invalid ID crashes the whole server. Please make sure to test more thoroughly your routes when you make changes. 

![image](https://user-images.githubusercontent.com/42552874/130660206-826b70c3-c322-4d63-bffb-76d1ee3e3bfb.png)

Reply:
Maybe the middleware that catches these errors could just forward the error instead of crashing the server? Here for example messaging return 400 for invalid request. Botpress could just send that status back instead of crashing.

I can also put a try catch around every call to messaging but it's a bit tedious

Reply:
At the very least it could return 500 and not crash the server

Reply:
We wrap every request with asyncMiddleware for that exact purpose. The issue is the logic added to the assertion middleware which is executed before the rest of the method (and the async mw). It needs to be wrapped in a try/catch and call next with an error if it fails

Reply:
Ah ok I see